docs/content/docs/HA.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "HA Design considerations"
+description: "Open Service Mesh HA Design considerations"
+type: docs
+---
+# HA Design considerations
+
+OSM is built with High Availability and Fault Tolerance in mind. The following sections will thoroughly document how these are tackled in OSM.
+
+## High Availability and Fault Tolerance
+
+High Availability and Fault Tolerance are implemented and ensured by several design decisions and external mechanisms in OSM, which will be documented in the following points:
+
+### Statelessness
+OSM does not own or have any state-dependent data that needs to be saved at runtime; with the controlled exceptions of:
+- Root certificate: when no root certificate is provided and Tresor is used, OSM will ensure atomicity and idempotence when creating the root certificate through Kubernetes secret API, even when multiple instances are spawned.
+_Note: it is required and always assumed the root certificate is the same for all control plane components._
```

Comment:
I didn't call it a guarantee because there's room for misconfiguration in - for example - cert-manager. Since providing the root cert can sometimes depend on the user, I think two instances can still run with a separate root if the root CA is manually renewed and new instances spawn with a new root CA.

I think code can be put in place to enforce and guarantee in most circumstances that if the secret is changed, we reload it, but I believe code is not there right now (is it?).

Reply:
It should be a guarantee as long as one follows the prerequisites to provision to a certificate provider to use with OSM. If you misconfigure this, pods will not even connect to the controller.